### ISSUE TITLE
Take a look to see if we should use Syscalls instead of CGO for Windows

### ISSUE THREAD

**Issue by BrentFarris** [KaijuEngine/kaiju#271]
Currently CGO is nice because it can have some consistency between platforms. Otherwise, [syscall](https://github.com/oakmound/shiny/blob/master/driver/internal/win32/zsyscall_windows.go#L86) can be used to construct and interact with windows.

**Comment by MatejMagat305** [comment]
If I can, I would like to notice, there is purego from ebiten, which call dlopen,  pass cgo funcs without compile cgo...

**Comment by BrentFarris** [comment]
> If I can, I would like to notice, there is purego from ebiten, which call dlopen, pass cgo funcs without compile cgo...

Thanks! I'll have to take a look at how they've got it setup.

**Comment by BrentFarris** [comment]
In relation to syscall.SyscallN, I found it to be [roughly 10% slower](https://twitter.com/KaijuCoder/status/1800264566397620277) than CGO on small functions.

**Comment by MatejMagat305** [comment]
> In relation to syscall.SyscallN, I found it to be [roughly 10% slower](https://twitter.com/KaijuCoder/status/1800264566397620277) than CGO on small functions.

it is true, however golang is loved for simply compilation, cgo is therefore problem that it need gcc and C dependency and hard cross-platform ..., beginer have problem on windows, so yes it is slow, but for beginer and easy cross is more suitable syscall ...., even purego is 5-10x slower than cgo, but it is able achieve 1396 fps (at basic example: https://github.com/gen2brain/raylib-go/pull/361), so at this case I as programer prefer easy cross, because usualy we need 60,120 fps

**Comment by BrentFarris** [comment]
Yeah, I'm leaving the branch with the changes up for further investigation; specifically for the initial compile time and the cross-compile benefits. I've not abandoned the idea, but do want to make sure I get all the measurements.

Something that would be pretty nice would be to have a compile tag to select which option you'd prefer, cgo for speed, or purego for ease of use.
